chrome/content/zotero/xpcom/zotero.js

```diff
@@ -381,7 +381,26 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 					return false;
 				}
 			}
-			
+			if(Zotero.File.isDropboxDirectory(dataDir.path) && Zotero.Prefs.get('warnDropboxDataDir')){
```

Comment:
Pref is only for the startup warning. Setting from prefs doesn't have the checkbox.

We could just always show at startup, but in theory if you really know what you're doing and don't have another computer it's OK.